src/main/java/com/example/spring_bank_project/Account/Domain/ValueObject/BirthDate.java

```diff
@@ -0,0 +1,24 @@
+package com.example.spring_bank_project.Account.Domain.ValueObject;
+
+import com.example.spring_bank_project.Account.Domain.Exception.UserIsNotAdultException;
+import lombok.Getter;
+import org.jetbrains.annotations.NotNull;
+
+import java.time.LocalDate;
+
+@Getter
+public class BirthDate {
+    private final LocalDate birthDate;
+
+    public BirthDate(String birthDate) {
+        LocalDate parsedBirthDate = LocalDate.parse(birthDate);
+        this.assertIsAdult(parsedBirthDate);
+        this.birthDate = parsedBirthDate;
+    }
+
+    private void assertIsAdult(@NotNull LocalDate birthDate) throws UserIsNotAdultException {
+        if (!birthDate.isBefore(LocalDate.now().minusYears(18))) {
```

Comment:
skoro ta metoda jest private i i tak nie mozesz jej uzyc poza scopem obiektu to po co przekazywac birthDate jako param? nie lepiej sie odwolac do this.birthDate?

Reply:
Dodatkowo LocalDate.now zaszyte na sztywno utrudni testowanie